runtime/runtime/src/verifier.rs

```diff
@@ -144,23 +146,45 @@ pub fn verify_and_charge_transaction(
         )?;
     let transaction = &signed_transaction.transaction;
     let signer_id = &transaction.signer_id;
+    let public_key = &transaction.public_key;
 
     let mut signer = match get_account(state_update, signer_id)? {
         Some(signer) => signer,
         None => {
             return Err(InvalidTxError::SignerDoesNotExist { signer_id: signer_id.clone() }.into());
         }
     };
-    let mut access_key = match get_access_key(state_update, signer_id, &transaction.public_key)? {
+    let mut access_key = match get_access_key(state_update, signer_id, public_key)? {
         Some(access_key) => access_key,
-        None => {
-            return Err(InvalidTxError::InvalidAccessKeyError(
-                InvalidAccessKeyError::AccessKeyNotFound {
-                    account_id: signer_id.clone(),
-                    public_key: transaction.public_key.clone(),
-                },
-            )
-            .into());
+        None => match signer_id.get_account_type() {
+            AccountType::EthImplicitAccount => {
+                if derive_account_id_from_public_key(public_key) == *signer_id {
+                    // TODO What about increasing storage usage for that account, because we added access key?
```

Comment:
We created ETH-implicit account without access key, that makes is cheaper regarding storage.
But now we add access key to the account. Should we handle that somehow, or it is handled automatically?